### PR TITLE
Add applicable integration test

### PR DESCRIPTION
**Pull request by attiasas**
- [x] All [tests](https://github.com/jfrog/jfrog-vscode-extension#building-and-testing-the-sources) passed. If this feature is not already covered by the tests, I added new tests.
- [x] I used `npm run format` for formatting the code before submitting the pull request.
-----

Add integration tests to download latest analyzerManager and run applicability scan on each package type.
Add cron to run all tests
Change to pull_request_target and starting to use 'save to test' to run tests

### CODE REVIEW

File: src/test/tests/utils/testIntegration.test.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    private _connectionManager!: ConnectionManager;
+
+    public static directory: string;
+
+    public async initialize() {
+        // Initialize connection manager
+        this._connectionManager = await createTestConnectionManager(this.logManager);
+        // Don't override existing connection details
+        let cacheUrl: string | undefined = process.env[ConnectionManager.URL_ENV];
+        let cacheToken: string | undefined = process.env[ConnectionManager.ACCESS_TOKEN_ENV];
```
Comment (yahavi) [inline src/test/tests/utils/testIntegration.test.ts:25]: Cache is something else
```suggestion
        let oldUrl: string | undefined = process.env[ConnectionManager.URL_ENV];
        let oldToken: string | undefined = process.env[ConnectionManager.ACCESS_TOKEN_ENV];
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -28,26 +36,42 @@ jobs:
         with:
           node-version: "16"
           check-latest: true
+
       - name: Install
         run: npm i --ignore-scripts
-        # Make sure no new dependencies were installed after running 'npm i'
+        
+      # Make sure no new dependencies were installed after running 'npm i'
+
```
Comment (yahavi) [inline .github/workflows/test.yml:44]: The comment is actually related to the below step. Let's remove the empty line.
```suggestion
```

File: src/test/tests/integration/applicabilityIntegration.test.ts [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+                assert.includeDeepMembers(Object.keys(response.applicableCve), Object.keys(expectedContent.applicableCve));
+            });
+
+            describe('Applicable details data validations', () => {
+                let expectedApplicableCves: Map<string, CveApplicableDetails>;
+                let responseApplicableCves: Map<string, CveApplicableDetails>;
+
+                before(() => {
+                    expectedApplicableCves = new Map<string, CveApplicableDetails>(Object.entries(expectedContent.applicableCve ?? []));
+                    assert.isTrue(expectedApplicableCves.size > 0);
```
Comment (yahavi) [inline src/test/tests/integration/applicabilityIntegration.test.ts:71]:
```suggestion
                    assert.isNotEmpty(expectedApplicableCves);
```

**Review by yahavi — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -1,9 +1,16 @@
...
+  push:
+  # Triggers the workflow on labeled PRs only.
+  pull_request_target:
+    types: [ labeled ]
+  schedule:
+    - cron: '0 6 * * *' # Runs every day at 6:00 AM GMT
 
 jobs:
   test:
+    if: github.event_name == 'schedule' || github.event_name == 'push' || (github.event_name == 'pull_request' && contains(github.event.pull_request.labels.*.name, 'safe to test'))
```
Comment (yahavi) [inline .github/workflows/test.yml:13]: Prevent running schedule on forks
```suggestion
    if: (github.event_name == 'schedule' && github.repository_owner == 'jfrog') || github.event_name == 'push' || (github.event_name == 'pull_request' && contains(github.event.pull_request.labels.*.name, 'safe to test'))
```

File: src/test/index.ts [hunk 5]
```diff
@@ -3,20 +3,22 @@ import { existsSync, writeFileSync } from 'fs-extra';
...
-export function run(): Promise<void> {
+export async function run(): Promise<void> {
     // Create the mocha test
     const mocha: Mocha = new Mocha({
         color: true,
         timeout: 600000
     } as MochaOptions);
 
     deleteWebviewJs();
+    await initializeIntegrationTests();
```
[the remark this answers is not in the document]
Comment (attiasas) [reply src/test/index.ts:19]: initializeIntegrationTests - downloads the AnalyzerManager once at the start.
integrationManager.initialize() - starts integration tests of specific scan that uses the analyzerManager we downloaded.
The directory needs to be static so we can access it in multiple files (done in the integrationManager.initialize())

**Pull request merged**